js/client.js

```diff
@@ -913,8 +935,8 @@
 				var nlIndex = data.indexOf('\n');
 				if (nlIndex > 0) {
 					this.receive(data.substr(nlIndex + 1));
-					nlIndex = parts[3].indexOf('\n');
-					parts[3] = parts[3].substr(0, nlIndex);
+					nlIndex = parts[4].indexOf('\n');
```

Comment:
Wait, this isn't backward compatible!

Reply:
Wait, the server change that's already in will break the client!

Reply:
Okay, I fixed it but it's a merge conflict. Your code will also need to be updated for backward compatibility.

Reply:
Alright, I'm just hiding the new toggles if the server doesn't have the new protocol.